### DISCUSSION TITLE
Unable to get type inference working with wagmi 0.7.0

### DISCUSSION THREAD

**Discussion by alecananian** [wevm/wagmi#1078] [Q&A] [answered]
I'm looking forward to being able to use the type inference features of wagmi 0.7.0, but upon upgrading I'm not seeing any changes. Package versions:

- wagmi: 0.7.0
- @wagmi/core: 0.6.0
- typescript: 4.8.4 (`strict` option set to `true`)

I've pasted the "love" function example [from the docs](https://wagmi.sh/docs/typescript#contract-abis) into my project and not getting type inference in Visual Studio Code:

<img width="559" alt="Screen Shot 2022-10-13 at 1 11 32 PM" src="https://user-images.githubusercontent.com/1013230/195700728-00451f36-fb41-48ac-90a3-a246c602c13c.png">

Any other steps to get this working that I'm missing before I create a reproducible example and issue?

**Comment by jxom** [comment]
Example should just work out of the box! Does this happen in a CodeSandbox also?

**Reply by alecananian** [reply]
Looks like it's not working in a CodeSandbox either:

<img width="559" alt="Screen Shot 2022-10-13 at 1 52 40 PM" src="https://user-images.githubusercontent.com/1013230/195708408-d7fcbc9e-e024-4757-933e-f0ab9433f21f.png">

https://codesandbox.io/s/dreamy-kate-ii14ev?file=/src/MyPage.tsx

**Reply by zkSoju** [reply]
I am also not getting any type inferences when migrating, updating typescript and wagmi, and using const assertion.

**Reply by tmm** [reply]
Fixed in `0.7.2` by https://github.com/wagmi-dev/wagmi/pull/1080

**Reply by zkSoju** [reply]
> Fixed in `0.7.2` by #1080

Works now!

**Reply by swheel33** [reply]
I am also unable to get the example to work even with 0.7.2 and it appears that the sandbox linked above is still not recognizing types even though the dependencies are updated to 0.7.2. Any thoughts?

**Reply by tmm** [reply]
@swheel33 happy to take a look if you share some code.

**Reply by alecananian** [reply]
@swheel33 agreed, I'm not seeing the type inference work in CodeSandbox or Visual Studio Code using 0.7.2 and 0.7.3

> @swheel33 happy to take a look if you share some code.

@tmm CodeSandbox linked above has been updated to 0.7.3: https://codesandbox.io/s/dreamy-kate-ii14ev?file=/src/MyPage.tsx

Edit: Though I don't know exactly which version of TypeScript is being used on CodeSandbox.
If it's what is shown in `package.json` here, it's too old: https://github.com/codesandbox/codesandbox-client/blob/master/package.json#L109

**Reply by alecananian** [reply]
I was able to get type inference working by right-clicking and going to definitions for a lot of different hooks/response types to force Visual Studio Code to refresh it all. The only one from the example that is not populating for me at the moment is the `const { data }` response, but getting close 😁

**Reply by swheel33** [reply]
@tmm Upon further digging the culprit was actually my ethers version was out of date, have to make a habit of updating all the dependent packages. Appreciate your help though especially on a saturday

**Comment by tmm** [comment, marked as answer]
https://github.com/wagmi-dev/wagmi/discussions/1078#discussioncomment-3874149
